### PR TITLE
Added unpaid payment calculator so unpaid workshops still generate su…

### PR DESCRIPTION
**Pull request by aoby**
…mmaries and show up in reports

### CODE REVIEW

File: dashboard/test/controllers/api/v1/pd/workshop_summary_report_controller_test.rb [hunk 1]
```diff
@@ -192,6 +192,21 @@ class Api::V1::Pd::WorkshopSummaryReportControllerTest < ::ActionController::Tes
...
+  test 'includes unpaid workshops' do
+    unpaid_workshop = create :pd_ended_workshop, organizer: @organizer, course: Pd::Workshop::COURSE_CSD
+    create :pd_workshop_participant, workshop: @workshop, enrolled: true, in_section: true, attended: true
+
+    sign_in @admin
+    get :index
+    assert_response :success
+    response = JSON.parse(@response.body)
+    assert_equal 3, response.count
+    unpaid_report = response.find{|row| row['workshop_id'] == unpaid_workshop.id}
```
[the remark this answers is not in the document]
Comment (aoby) [reply dashboard/test/controllers/api/v1/pd/workshop_summary_report_controller_test.rb:204]: Hound! At least they "notified" us this time... https://twitter.com/houndci/status/811271775904333825

File: dashboard/lib/pd/payment/payment_calculator_unpaid.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+module Pd::Payment
+  # Calculator for unpaid workshops. This generates a summary with zero payment.
```
Comment (mehalshah) [inline dashboard/lib/pd/payment/payment_calculator_unpaid.rb]: This seems to be more of a nil payment than a zero payment

File: dashboard/lib/pd/payment/payment_factory.rb [hunk 3]
```diff
@@ -27,7 +27,7 @@ def self.get_calculator_class(workshop)
         ].include?(workshop.course)
       end
 
-      nil
+      PaymentCalculatorUnpaid
```
Comment (mehalshah) [inline dashboard/lib/pd/payment/payment_factory.rb:30]: Is there a test for this that verifies that paymentcalculatorunpaid is what it falls back to?

Comment (aoby) [reply dashboard/lib/pd/payment/payment_factory.rb:30]: Yes, for 2 specific examples: https://github.com/code-dot-org/code-dot-org/pull/12651/files#diff-2b6b6c2c760761d8b2e4668eae8dc74eR56

I could add a test with a mocked made up course we know will never match to verify the fallback.

File: dashboard/lib/pd/payment/payment_calculator_unpaid.rb (hunk 2 above)
Comment (aoby) [reply dashboard/lib/pd/payment/payment_calculator_unpaid.rb]: Fair point. Will update the comment.

Comment (aoby) [reply dashboard/lib/pd/payment/payment_calculator_unpaid.rb]: How about "with no payment"?

File: dashboard/lib/pd/payment/payment_factory.rb (hunk 3 above)
Comment (aoby) [reply dashboard/lib/pd/payment/payment_factory.rb:30]: Added a test [workshops that do not match a known payment type fall through to unpaid](https://github.com/code-dot-org/code-dot-org/pull/12651/commits/0ba995df1f7d0b6986382431d80828be04399aaf#diff-2b6b6c2c760761d8b2e4668eae8dc74eR67)

**Comment by aoby** [comment]
The hound error is now erroneous and can be ignored. That rule is disabled in [.rubocop_todo.yml](https://github.com/code-dot-org/code-dot-org/blob/637f00711d4fc6f09663d40c78c7bef980b3a933/.rubocop_todo.yml#L135-L135), and hound has an open [issue](https://github.com/houndci/hound/issues/1290) that it's not honoring included rubocop config files.

**Pull request merged**